src/verifact_agents/evidence_hunter.py

```
@@ -10,46 +18,226 @@
     relevance: float = 1.0
     stance: str = "supporting"  # supporting, contradicting, neutral
+    credibility: float = 1.0
+    timestamp: str = Field(default_factory=lambda: datetime.now().isoformat())
+
+def deduplicate_evidence(evidence_list):
+    """Deduplicate evidence list by source and content.
+    Args:
+        evidence_list (list[Evidence]): The list of evidence to deduplicate.
+    Returns:
+        list[Evidence]: The deduplicated list of evidence.
+    """
```

Ruff (0.11.9)

1 blank line required between summary line and description

(D205)